File: starterpack/microchallenge/java/src/main/java/com/coveo/MyBot.java
package com.coveo;

import io.javalin.Javalin;
import picocli.CommandLine;
import picocli.CommandLine.Option;

public class MyBot
{
  public static class Args {
    @Option(names = { "-p", "--port" },
        description = "set port number (default: 8010)",
        defaultValue = "8010")
    public int port;
  }

  public static void main (String [] args) {
    Args arguments = new Args();
    new CommandLine(arguments).parse(args);

    Javalin app = Javalin.create().start(arguments.port);
    app.get("/microchallenge", ctx -> {
      
      System.out.println("\n\n\n-------------------------- REQUEST LOGS STARTING HERE --------------------------");
      System.out.println("You can log stuff and download the logs from the UI in the replay section.");
      System.out.println("Here is the current problem:");
      System.out.println(ctx.queryParam("problem")); // problem is in json format
      System.out.println("---------------------------------------------------------------------------------");
      
      ctx.result(Integer.toString(18));
    });
  }

}

File: starterpack/microchallenge/csharp/MyBot.cs
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;
using System;

namespace Blitz_2019
{
    [Route("/")]
    [ApiController]
    public class MyBot : ControllerBase
    {
        [HttpGet("/microchallenge")]
        public ActionResult<string> Solve(string problem)
        {
            Console.WriteLine("\n\n\n-------------------------- REQUEST LOGS STARTING HERE --------------------------");
            Console.WriteLine("You can log stuff and download the logs from the UI in the replay section.");
            Console.WriteLine("Here is the current problem:");
            Console.WriteLine(problem); // problem is in json format
            Console.WriteLine("---------------------------------------------------------------------------------");

            return 18.ToString();
        }
    }
}
